packages/blade/src/components/Amount/Amount.tsx

```diff
@@ -0,0 +1,193 @@
+import { BaseText } from '~components/Typography/BaseText';
+import type { Feedback } from '~tokens/theme/theme';
+import type { BaseTextProps } from '~components/Typography/BaseText/types';
+import BaseBox from '~components/Box/BaseBox';
+import type { TestID } from '~src/_helpers/types';
+import { metaAttribute, MetaConstants } from '~utils';
+
+export const suffixTypes = {
+  HUMANIZE: 'humanize',
+  DECIMALS: 'decimals',
+  NONE: 'none',
+};
+
+type currencyType = 'INR' | 'MYR';
+
+type AmountProps = {
+  /**
+   * The value to be rendered within the component.
+   *
+   */
+  value: number;
+  /**
+   * Sets the intent of the amount.
+   *
+   * @default 'neutral'
+   */
+  intent?: Feedback | undefined;
+  /**
+   * Sets the variant of the amount.
+   *
+   * @default 'heading'
+   */
+  size?:
+    | `title-medium`
+    | `title-small`
+    | `heading-large`
+    | `heading-large-bold`
+    | `heading-small`
+    | `heading-small-bold`
+    | `body-medium`
+    | `body-medium-bold`
+    | `body-small`
+    | `body-small-bold`;
+  /**
+   * Indicates whether a text suffix should be used
+   *
+   * @default 'decimals'
+   */
+  suffix?: 'decimals' | 'none' | 'humanize';
+  /**
+   * Highlight the main amount by making the prefix symbol and decimal digits small
+   *
+   * @default true
+   */
+  isAffixSubtle?: true | false;
+  /**
+   * Prefix to be shown before the currency value. The prefix can be either a currency symbol or a currency code.
+   *
+   * @default 'currency-symbol'
+   */
+  prefix?: 'currency-symbol' | 'currency-code';
+} & TestID;
+
+type ColorProps = {
+  textColor: BaseTextProps['color'];
+  prefixSuffixColor: BaseTextProps['color'];
+};
+
+const getColorProps = ({ intent }: { intent: AmountProps['intent'] }): ColorProps => {
+  const props: ColorProps = {
+    textColor: 'surface.text.normal.lowContrast',
+    prefixSuffixColor: 'surface.text.muted.lowContrast',
+  };
+  if (!intent) return props;
+  props.textColor = `feedback.text.${intent}.lowContrast`;
+  props.prefixSuffixColor = `feedback.text.${intent}.lowContrast`;
+  if (intent === 'neutral') {
+    props.prefixSuffixColor = `surface.text.muted.lowContrast`;
+  }
+  return props;
+};
+
+const getFlooredFixed = (value: number, precision: number): number => {
+  const factor = 10 ** precision;
+  const roundedValue = Math.floor(value * factor) / factor;
+  return Number(roundedValue.toFixed(precision));
+};
+
+const addCommas = (num: number, currency: currencyType): string => {
+  const locale = currencylocaleMapping[currency];
+  return num.toLocaleString(locale);
+};
+
+const getFormattedAmountWithSuffixSymbol = (num: number, currency: currencyType): string => {
+  const abbreviations: { value: number; symbol: string }[] = currencyAbbreviationsMapping[currency];
+
+  const abbreviation = abbreviations.find((abbr) => num >= abbr.value);
+  if (abbreviation) {
+    num = num / abbreviation.value;
+    const formattedNum = Number(getFlooredFixed(num, 2));
```

Comment:
Verbose variable name. Also, why do we need to wrap it in `Number`? Doesn't getFloorFixed always return number?
```suggestion
    const formattedAmountValue = getFlooredFixed(num, 2);
```